File: melius-backend/servicios/proveedor_service.py
from repositorio import proveedor_repository

def listar():
    return proveedor_repository.listar()

def crear(p):
    return proveedor_repository.crear(p)

def actualizar(p):
    return proveedor_repository.actualizar(p)

def eliminar(id):
    return proveedor_repository.eliminar(id)

File: Programa_Final_Frontend/src/app/services/proveedor.service.ts
import { Injectable } from '@angular/core';
import { BaseHttpService } from './base-http.service';
import { Proveedor } from '../models/proveedor.model';

@Injectable({ providedIn: 'root' })
export class ProveedorService {
  constructor(private http: BaseHttpService) {}

  listar(): Promise<Proveedor[]> {
    return this.http.get<Proveedor[]>('/proveedores/');
  }

  crear(proveedor: Proveedor): Promise<Proveedor> {
    return this.http.post<Proveedor>('/proveedores/', proveedor);
  }

  actualizar(proveedor: Proveedor): Promise<Proveedor> {
    return this.http.put<Proveedor>(`/proveedores/${proveedor.id}`, proveedor);
  }

  eliminar(id: number): Promise<void> {
    return this.http.delete<void>(`/proveedores/${id}`);
  }
}
